### ISSUE TITLE
StreamPeerTCP.is_connected() always return true 

### ISSUE THREAD

**Issue by SuperUserNameMan** [godotengine/godot#4675]
**Operating system or device - Godot version:**
Official 2.0.3.stable win64, running under Windows 8.1

**Issue description** (what happened, and what was expected):

Once `StreamPeerTCP.connect()` is called, `StreamPeerTCP.is_connected()` will always return true even if there is no server running at the other side.

This behavior makes `.is_connected()` useless to know if it is actually connected to a server.

**Steps to reproduce:**
Into the example below, we want to try to connect to the server every 3 seconds.
But it does not work because client.is_connected() return true as soon as client.connect() is called, this even if there is no server running.

**Link to minimal example project** (optional but very welcome):
````
extends Control

const PORT = 64335
const HOST = "127.0.0.1"
var client = StreamPeerTCP.new()
var server_replied = false

const TIMEOUT = 3

var timeout_counter = 0
var connect_attemps_counter = 0

func _ready():
	set_process(true)
	pass
	
func _process(delta):
	if not client.is_connected() :
		if timeout_counter == 0 :
			connect_attemps_counter += 1
			print("Connecting to ",HOST,":",PORT," (",connect_attemps_counter,")")
			client.connect(HOST,PORT)
		else:
			timeout_counter += delta
			if timeout_counter >= TIMEOUT :
				timeout_counter = 0 # we would like to try to connect again
	elif client.is_connected() and client.get_available_bytes()>0 :
		var r = client.get_var()
		server_replied = true
		print("Connected")
	pass

````

**Comment by Kermer** [comment]
I think you wanted to check `StreamPeerTCP.get_status() == STATUS_CONNECTED`.
I think that the main and almost only point of `is_connected` is to check it before using `disconnect` so it don't display errors.
[I just stick to checking the status](http://docs.godotengine.org/en/latest/classes/class_streampeertcp.html?highlight=streampeertcp#numeric-constants).

You should also use [PacketPeerStream](http://docs.godotengine.org/en/latest/classes/class_packetpeerstream.html) together with StreamPeerTCP

**Comment by SuperUserNameMan** [comment]
Yes you're right. get_status() seems to be what I wanted. I'll rewrite my code using it. thanks :-)

And maybe I misinterpreted the role of .is_connected(). If so, this is issue might not be related to a bug, but to a lack of documentation.

**Comment by J08nY** [comment]
Should be fixed by  #5216
